samples/objc/FirebaseUI-demo-objc/Samples/Auth/FUIAuthViewController.m

```diff
@@ -344,31 +354,40 @@ - (void)showAlertWithTitlte:(NSString *)title message:(NSString *)message {
 
 }
 
-+ (NSArray *)getAllIDPs {
-  NSArray<NSIndexPath *> *selectedRows = @[
-    [NSIndexPath indexPathForRow:kIDPEmail inSection:kSectionsProviders],
-    [NSIndexPath indexPathForRow:kIDPGoogle inSection:kSectionsProviders],
-    [NSIndexPath indexPathForRow:kIDPFacebook inSection:kSectionsProviders],
-    [NSIndexPath indexPathForRow:kIDPTwitter inSection:kSectionsProviders],
-    [NSIndexPath indexPathForRow:kIDPPhone inSection:kSectionsProviders],
-    [NSIndexPath indexPathForRow:kIDPAnonymous inSection:kSectionsProviders]
-  ];
-  return [self getListOfIDPs:selectedRows useCustomScopes:NO];
-}
-
 - (NSArray *)getListOfIDPs {
-  return [[self class] getListOfIDPs:[self.tableView indexPathsForSelectedRows] useCustomScopes:_customScopeSwitch.isOn];
+  return [[self class] getListOfIDPs:[self.tableView indexPathsForSelectedRows]
+                     useCustomScopes:_customScopeSwitch.isOn
+                        useEmailLink:_emailSwitch.isOn];
 }
 
-+ (NSArray *)getListOfIDPs:(NSArray<NSIndexPath *> *)selectedRows useCustomScopes:(BOOL)useCustomScopes {
++ (NSArray *)getListOfIDPs:(NSArray<NSIndexPath *> *)selectedRows
+           useCustomScopes:(BOOL)useCustomScopes
+              useEmailLink:(BOOL)useEmaiLink {
   NSMutableArray *providers = [NSMutableArray new];
 
   for (NSIndexPath *indexPath in selectedRows) {
     if (indexPath.section == kSectionsProviders) {
       id<FUIAuthProvider> provider;
       switch (indexPath.row) {
         case kIDPEmail:
-          provider = [[FUIEmailAuth alloc] init];
+          if (useEmaiLink) {
+            // ActionCodeSettings for email link sign-in.
+            FIRActionCodeSettings *actionCodeSettings = [[FIRActionCodeSettings alloc] init];
+            [actionCodeSettings setURL:[NSURL URLWithString:@"https://fb-sa-1211.appspot.com"]];
```

Comment:
this is  property of FIRActionCodeSettings. no need to use setURL. use property syntax.